nav2_safety_nodes/src/nav2_safety_node.cpp

```diff
@@ -0,0 +1,229 @@
+  declare_parameter("zone_priority", rclcpp::ParameterValue(1));
+  declare_parameter("zone_num_pts", rclcpp::ParameterValue(1));
+  declare_parameter("base_frame", std::string("base_link"));
+  declare_parameter("tf_tolerance", rclcpp::ParameterValue(0.01));
+  declare_parameter("scan_topics", scan_topics);
+  declare_parameter("speed_limit_topic", rclcpp::ParameterValue("speed_limit"));
+}
+
+nav2_util::CallbackReturn
+SafetyZone::on_configure(const rclcpp_lifecycle::State & /*state*/)
+{
+  RCLCPP_INFO(logger_, "Configuring and Setting up");
+
+  // Get all of the required parameters
+  safety_polygon_ = get_parameter("safety_polygon").as_string();
+  update_frequency_ = get_parameter("update_frequency").as_double();
+  zone_action_ = get_parameter("zone_action").as_double();
+  zone_priority_ = get_parameter("zone_priority").as_int();
+  zone_num_pts_ = get_parameter("zone_num_pts").as_int();
+  base_frame_ = get_parameter("base_frame").as_string();
+  tf_tolerance_ = get_parameter("tf_tolerance").as_double();
+  scan_topics_ = get_parameter("scan_topics").as_string_array();
+  speed_limit_topic_ = get_parameter("speed_limit_topic").as_string();
+
+  // If the safety_polygon has been specified, it must be in the correct format
+  if (safety_polygon_ != "" && safety_polygon_ != "[]") {
+    // Polygon parameter has been specified, polygon -> point vector(safety_zone)
+    nav2_util::makeVectorPointsFromString(safety_polygon_, safety_zone_);
+  } else {
+    throw std::runtime_error("The safety_polygon is invalid");
+  }
+
+  // Initialize transform listener and broadcaster
+  tf2_ = std::make_unique<tf2_ros::Buffer>(get_clock());
+  auto timer_interface = std::make_shared<tf2_ros::CreateTimerROS>(
+    get_node_base_interface(), get_node_timers_interface());
+  tf2_->setCreateTimerInterface(timer_interface);
+  tf2_listener_ = std::make_unique<tf2_ros::TransformListener>(*tf2_);
+
+  return nav2_util::CallbackReturn::SUCCESS;
+}
+
+nav2_util::CallbackReturn
+SafetyZone::on_activate(const rclcpp_lifecycle::State & /*state*/)
+{
+  RCLCPP_INFO(logger_, "Activating");
+  // Create the publishers and subscribers
+  safety_polygon_pub_ = create_publisher<geometry_msgs::msg::PolygonStamped>(
+    "safety_zone", rclcpp::SystemDefaultsQoS());
+  // Velocity publisher
+  speed_limit_pub_ = create_publisher<nav2_msgs::msg::SpeedLimit>(
+    speed_limit_topic_, rclcpp::QoS(10));
+  // Multiple Laserscan subscribers
+  scan_subscribers_.reserve(scan_topics_.size());
+  for (unsigned int i = 0; i < scan_topics_.size(); i++) {
+    RCLCPP_INFO(logger_, "Subscribing to %s topic", scan_topics_[i].c_str());
+    scan_subscribers_.push_back(
+      create_subscription<sensor_msgs::msg::LaserScan>(
+        scan_topics_[i].c_str(), rclcpp::SystemDefaultsQoS(),
+        std::bind(&SafetyZone::laserCallback, this, std::placeholders::_1)));
+  }
+  auto update_frequency_s_ = std::chrono::duration<double>(update_frequency_);
```

Comment:
`update_frequency_s_` should not have an underscore, it is not a class variable.